File: 0201_0300/0271/0271.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    n = int(f.readline())
    a, b, i = 1, 2, 3
    while b < n:
        a, b, i = b, a + b, i + 1
    q.write('0' if b > n else "1\n" + str(i))
        

File: 0201_0300/0271/0271.c
#include <stdio.h>

int main()
{
	FILE *f = fopen("input.txt", "r"), *q = fopen("output.txt", "w");
	int n, a, b = 1, c = 2, i = 3;

	fscanf(f, "%d", &n);
	while (c < n)
		a = b, b = c, c = a + b, ++i;
	fprintf(q, c > n ? "0" : "1\n%d", i);
	return 0;
}
